dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2625,190 @@ void FileSprayer::pushParts()
+            if (targetType==StorageType::StorageTypeAzureFile)
+            {
+                auto fileClient = Shares::ShareFileClient(targetURI.str());
+                fileCopyOp.emplace_back(std::move(fileClient.StartCopy(sourceURI.str())));
+            }
+            else
+            {
+                auto blobClient = BlobClient(targetURI.str());
+                blobCopyOp.emplace_back(std::move(blobClient.StartCopyFromUri(sourceURI.str())));
+            }
+            copyingInProgress.emplace_back(idx);
+            curprogress.status =  OutputProgress::StatusBegin;
+            curprogress.numReads++;
+            curprogress.numWrites++;
+            if (isAborting())
+                break;
+         }
+    }
+    catch (const Azure::Core::RequestFailedException& e)
+    {
+        IERRLOG("transferUsingAPI failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+        return false;
+    }
+    // Track progress of copy operations
+    while (!isAborting())
+    {
+        bool anyRemaining = false;
+        auto itProgress = std::begin(copyingInProgress);
+        if (targetType==StorageType::StorageTypeAzureFile)
+        {
+            for(auto & f: fileCopyOp)
+            {
+                OutputProgress & curprogress = progress.item(*itProgress);
+                ++itProgress;
+                if (curprogress.status==OutputProgress::StatusCopied)
+                    continue;
+                try
+                {
+                    auto response = &f.Poll();
+                    auto props = f.Value();
+                    auto status = props.CopyStatus.HasValue()?props.CopyStatus.Value():(Shares::Models::CopyStatus::Pending);
+                    curprogress.outputLength = props.FileSize;
+                    curprogress.resultTime.setString(props.LastModified.ToString().c_str());
+                    if (status==Shares::Models::CopyStatus::Success)
+                    {
+                        curprogress.status = OutputProgress::StatusCopied;
+                        curprogress.outputCRC = curprogress.inputCRC;
+                    }
+                    else if (status==Shares::Models::CopyStatus::Pending)
+                    {
+                        curprogress.status =  OutputProgress::StatusActive;
+                        anyRemaining=true;
+                    }
+                    else
+                        return false; //Aborted or Failed -> return false to use older copy method
+                }
+                catch(const Azure::Core::RequestFailedException& e)
+                {
+                    IERRLOG("transferUsingAPI .poll() failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+                    return false;
+                }
+                if (isAborting())
+                    break;
+            }
+        }
+        else
+        {
+            for(auto & b: blobCopyOp)
```

Comment:
The bulk of this is similar to the code above - should be shared if at all possible.
Probably cleaner as an array of copy objects with derived classes for the different cases.